tools/autograd/gen_variable_type.py

```diff
@@ -485,11 +485,32 @@ def is_differentiable(arg):
         if 'Tensor' not in arg['type']:
             return False
         if arg['dynamic_type'] in {'IndexTensor', 'BoolTensor'}:
+            # TODO: Enable this after native_functions.yaml schema unification.
+            # These are necessary for legacy code and should be
+            # used by legacy code only!
+            # assert name.startswith('_th_'), \
```

Comment:
isn't there a mode or something you can check?

Reply:
Maybe, I'll revisit this once IndexTensor/BoolTensor has been entirely removed. Until then this needs to exist as a general reminder.

Reply:
sound good.  I think checking that the mode isn't 'TH' and isn't 'NN" is sufficient.